src/Birkhoff_polytope.jl

```diff
@@ -0,0 +1,453 @@
+    if sense == :greaterthan
+        blmo.lower_bounds[c_idx] = value
+    elseif sense == :lessthan
+        blmo.upper_bounds[c_idx] = value
+    else
+        error("Allowed values for sense are :lessthan and :greaterthan.")
+    end
+end
+
+# Delete bounds.
+function Boscia.delete_bounds!(blmo::BirkhoffBLMO, cons_delete)
+    for (d_idx, sense) in cons_delete
+        if sense == :greaterthan
+            blmo.lower_bounds[d_idx] = -Inf
+        else
+            blmo.upper_bounds[d_idx] = Inf
+        end
+    end
+end
+
+# Add bound constraint.
+function Boscia.add_bound_constraint!(blmo::BirkhoffBLMO, key, value, sense::Symbol)
+    idx = findfirst(x -> x == key, blmo.int_vars)
+    if sense == :greaterthan
+        blmo.lower_bounds[idx] = value
+    elseif sense == :lessthan
+        blmo.upper_bounds[idx] = value
+    else
+        error("Allowed value of sense are :lessthan and :greaterthan!")
+    end
+end
+
+## Checks
+
+# Check if the subject of the bound c_idx is an integer variable (recorded in int_vars).
+function Boscia.is_constraint_on_int_var(blmo::BirkhoffBLMO, c_idx, int_vars)
+    return blmo.int_vars[c_idx] in int_vars
+end
+
+# To check if there is bound for the variable in the global or node bounds.
+function Boscia.is_bound_in(blmo::BirkhoffBLMO, c_idx, bounds)
+    return haskey(bounds, blmo.int_vars[c_idx])
+end
+
+# Has variable an integer constraint?
+function Boscia.has_integer_constraint(blmo::BirkhoffBLMO, idx)
+    return idx in blmo.int_vars
+end
+
+## Optional
+
+function Boscia.check_feasibility(blmo::BirkhoffBLMO)
+    for (lb, ub) in zip(blmo.lower_bounds, blmo.upper_bounds)
+        if ub < lb
+            return Boscia.INFEASIBLE
+        end
+    end
+    # For double stochastic matrices, each row and column must sum to 1
+    # We check if the bounds allow for feasible assignments
+    n0 = blmo.dim
+    n = n0^2
+    # Initialize row and column bound tracking
+    row_min_sum = zeros(n)  # minimum possible sum for each row
+    row_max_sum = zeros(n)  # maximum possible sum for each row
+    col_min_sum = zeros(n)  # minimum possible sum for each column
+    col_max_sum = zeros(n)  # maximum possible sum for each column
+
+    # Process each integer variable
+    for idx in eachindex(blmo.int_vars)
+        var_idx = blmo.int_vars[idx]
+
+        # Convert linear index to (row, col) based on storage format
+        if blmo.append_by_column
+            j = ceil(Int, var_idx / n0)  # column index
+            i = Int(var_idx - n0 * (j - 1))  # row index
+        else
+            i = ceil(Int, var_idx / n0)  # row index  
+            j = Int(var_idx - n0 * (i - 1))  # column index
+        end
+
+        # Add bounds to row and column sums
+        row_min_sum[i] += blmo.lower_bounds[idx]
+        row_max_sum[i] += blmo.upper_bounds[idx]
+        col_min_sum[j] += blmo.lower_bounds[idx]
+        col_max_sum[j] += blmo.upper_bounds[idx]
```

Comment:
How is the convention here now? Do we store the bounds for all variables or just for the integer variables?